test/solidity/payment-channel/PaymentChannel.js

```diff
@@ -0,0 +1,136 @@
+/*-
+ *
+ * Hedera Smart Contracts
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const PaymentChannelHelper = require('./helper')
+const { GAS_LIMIT_1_000_000 } = require('../../constants')
+
+describe('@solidityequiv3 PaymentChannel', () => {
+  const GASLIMIT = 1000000
+  const DURATION = 3 // 3 seconds
+  const OWED_AMOUNT = 100000000
+  const INITIAL_FUND = ethers.utils.parseEther('3')
+  let signers,
+    senderAddress,
+    recipientAddress,
+    paymentSignature,
+    paymentChannelContract
+
+  before(async () => {
+    signers = await ethers.getSigners()
+    senderAddress = await signers[0].getAddress()
+    recipientAddress = await signers[1].getAddress()
+
+    const paymentChannelContractFactory = await ethers.getContractFactory(
+      'PaymentChannel'
+    )
+
+    paymentChannelContract = await paymentChannelContractFactory.deploy(
+      recipientAddress,
+      DURATION,
+      {
+        gasLimit: GASLIMIT,
+        value: INITIAL_FUND,
+      }
+    )
+
+    paymentSignature = await PaymentChannelHelper.signPayment(
+      signers[0],
+      paymentChannelContract.address,
+      OWED_AMOUNT
+    )
+  })
+
+  it('Should deployed with correct deployed arguments - open payment channel', async () => {
+    const contractBalance = await ethers.provider.getBalance(
+      paymentChannelContract.address
+    )
+
+    expect(contractBalance).to.eq(INITIAL_FUND)
+    expect(await paymentChannelContract.expiration()).to.not.eq(0)
+    expect(await paymentChannelContract.sender()).to.eq(senderAddress)
+    expect(await paymentChannelContract.recipient()).to.eq(recipientAddress)
+  })
+
+  it('Should close the payment channel when recipient execute close method', async () => {
+    const transaction = await paymentChannelContract
+      .connect(signers[1])
+      .close(OWED_AMOUNT, paymentSignature)
+
+    const receipt = await transaction.wait()
+
+    const [contractBaleBefore, senderBalBefore, recipientBalBefore] =
```

Comment:
Nit: Did you mean to call it 'contractBaleBefore'?  Not 'contractBalanceBefore'?

Reply:
Ohh nice catch! Definitely a typo! Thanks Eric! Meant to call it `contractBalBefore`